src/main/java/edu/harvard/iq/dataverse/engine/command/impl/LinkDatasetCommand.java

```diff
@@ -50,6 +52,13 @@ public DatasetLinkingDataverse execute(CommandContext ctxt) throws CommandExcept
         if (linkedDataset.getOwner().getOwners().contains(linkingDataverse)) {
             throw new IllegalCommandException(BundleUtil.getStringFromBundle("dataset.link.not.to.parent.dataverse"), this);
         }
+        List<Dataverse> existingLinksToDataverses = new ArrayList<>();
+        for (DatasetLinkingDataverse datasetLinkingDataverse : linkedDataset.getDatasetLinkingDataverses()) {
```

Comment:
will this be efficient enough, instead of adding a find method specific for an existing link?

Reply:
@scolapasta and I just discussed this and decided that we'd see how it goes in QA and optimize later if need be.

Reply:
Good news, I found an alreadyLinked method and switched to it in 109a3d3a1.